### PR TITLE
remove datetime.pxd

### PR DESCRIPTION
**Pull request by jbrockmendel**
There is an unrelated 2-line edit in hashing.pyx that gets rid of a compiler warning.

### CODE REVIEW

File: pandas/_libs/tslibs/np_datetime.pxd [hunk 1]
```diff
@@ -71,3 +75,6 @@ cdef int64_t pydate_to_dt64(date val, pandas_datetimestruct *dts)
 cdef npy_datetime get_datetime64_value(object obj) nogil
 cdef npy_timedelta get_timedelta64_value(object obj) nogil
 cdef PANDAS_DATETIMEUNIT get_datetime64_unit(object obj) nogil
+
+cdef int _string_to_dts(object val, pandas_datetimestruct* dts,
```
Comment (jreback) [inline pandas/_libs/tslibs/np_datetime.pxd]: can deprivatize at some point

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm ping on green

File: pandas/_libs/tslibs/np_datetime.pxd (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/np_datetime.pxd]: Yah I’m currently running asv on a proposed change to string_to_dts that would make to_datetime and Timestamp behavior more similar (or at least make the dissimilarity more obvious)

**Comment by jbrockmendel** [comment]
Ping

**Review by jreback — APPROVED** [review]

File: pandas/_libs/tslibs/np_datetime.pyx [hunk 2]
```diff
@@ -161,3 +167,35 @@ cdef inline int64_t pydate_to_dt64(date val,
     dts.hour = dts.min = dts.sec = dts.us = 0
     dts.ps = dts.as = 0
     return dtstruct_to_dt64(dts)
+
+
+cdef inline int _string_to_dts(object val, pandas_datetimestruct* dts,
+                               int* out_local, int* out_tzoffset) except? -1:
+    cdef:
```
Comment (jreback) [inline pandas/_libs/tslibs/np_datetime.pyx]: when you do change this, pls add doc-string (and same below)

File: pandas/_libs/tslibs/np_datetime.pyx [hunk 3]
```diff
@@ -161,3 +167,35 @@ cdef inline int64_t pydate_to_dt64(date val,
...
+
+    if result == -1:
+        raise ValueError('Unable to parse %s' % str(val))
+    return result
+
+
+cdef inline int _cstring_to_dts(char *val, int length,
+                                pandas_datetimestruct* dts,
+                                int* out_local, int* out_tzoffset) except? -1:
+    # Note: without this "extra layer" between _string_to_dts
```
Comment (jreback) [inline pandas/_libs/tslibs/np_datetime.pyx]: have to investigate this, it is pretty weird

**Comment by jreback** [comment]
thanks! pls make note of those 2 comments above on your list.

**Pull request merged**